src/main/java/edu/wpi/grip/ui/OperationView.java

```diff
@@ -88,5 +88,31 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
             Step step = new Step(this.eventBus, this.operation);
             this.eventBus.post(new StepAddedEvent(step));
         });
+
+        // Ensures that when this element is hidden that it also removes its size calculations
+        this.managedProperty().bind(this.visibleProperty());
+    }
+
+    // Cache the lower case name so that it isn't calculated each time the search bar is used
+    private String nameLoweCase = null;
+    /**
+     * Uses the search string from the palette and allows this element to determine if it should be visible
+     * @param lowerCaseSearchString The search string that should be used to determine visibility
+     */
+    void determineVisibility(String lowerCaseSearchString){
+        assert lowerCaseSearchString.equals(lowerCaseSearchString.toLowerCase()) : "The string passed to determine visibility was not lower case";
+        if (nameLoweCase == null) nameLoweCase = getOperationName().toLowerCase();
+        if (lowerCaseSearchString.isEmpty() || nameLoweCase.contains(lowerCaseSearchString)){
+            this.setVisible(true);
+        } else {
+            this.setVisible(false);
+        }
+    }
+
+    /*
+     * Intentionally left package protected
+     */
+    String getOperationName(){
```

Comment:
Better to just add a `public Operation getOperation()`.  We're probably going to do a lot more with sorting and filtering and stuff than just by name, and it would be nice to not have to add a method to this class every time we want to get a different thing from `Operation`.